setup.py

```diff
@@ -76,16 +75,46 @@ def write_version_file():
 pillow_req = 'pillow-simd' if get_dist('pillow-simd') is not None else 'pillow'
 requirements.append(pillow_req + pillow_ver)
 
+third_party_libraries = ['png', 'turbojpeg']
+third_party_dir = os.path.join(cwd, "third_party")
+third_party_lib_directories = ['libpng', 'libjpeg-turbo']
+third_party_search_directories = [os.path.join(third_party_dir, directory) for directory in third_party_lib_directories]
+
+def _build_cmake_dependency(directory, args=""):
+    os.chdir(directory)
+    if sys.platform == 'win32':
+        os.system("cmake3.exe --clean .")
+        os.system("cmake3.exe -G='Visual Studio 15 2017' {} .".format(args) )
+        os.system("cmake3.exe --build .")
+    else: 
+        os.system("cmake --clean .")
+        os.system("cmake {} .".format(args) )
+        os.system("cmake --build .")
+    os.chdir(cwd)
+
+
+def build_dependencies():
```

Comment:
[This](https://github.com/pytorch/pytorch/blob/master/setup.py#L286) is how PyTorch set this up in case you didn't see it. There might be some patterns that could be borrowed from this.

Reply:
To be fair, 
I am having a hard time especially with the windows CI and MacOs, because I did not use those systems in years. 
I am trying to get something to work ( even if it's dirty ). Then I'll clean things up for the merge.

Reply:
Hi, 
I took a look. I'll grab some code. 
Thanks